pkgs/by-name/sl/slepc/package.nix

```diff
@@ -0,0 +1,115 @@
+{
+  lib,
+  stdenv,
+  cctools,
+  fetchFromGitLab,
+  sowing,
+  python3,
+  python3Packages,
+  gfortran, # will be removed when pr 389036 is merged
+  arpack-mpi,
+  petsc,
+  mpi,
+  mpiCheckPhaseHook,
+  pythonSupport ? false,
+  withExamples ? false,
+  withArpack ? stdenv.hostPlatform.isLinux,
+}:
+assert petsc.mpiSupport;
+assert pythonSupport -> petsc.pythonSupport;
+stdenv.mkDerivation (finalAttrs: {
+  pname = "slepc";
+  version = "3.22.2";
+
+  src = fetchFromGitLab {
+    owner = "slepc";
+    repo = "slepc";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-a5DmsA7NAlhrEaS43TYPk7vtDfhXLEP+5sftu2A9Yt4=";
+  };
+
+  postPatch = ''
+    # fix slepc4py install prefix
+    substituteInPlace config/packages/slepc4py.py \
+      --replace-fail "slepc.prefixdir,'lib'" \
+      "slepc.prefixdir,'${python3.sitePackages}'"
+
+    patchShebangs lib/slepc/bin
+
+    # use system bfort
+    substituteInPlace config/packages/sowing.py \
+      --replace-fail "bfort = os.path.join(archdir,'bin','bfort')" \
+      "bfort = '${sowing}/bin/bfort'"
+  '';
```

Comment:
Here too I'd recommend using a `replaceVars` on a patch, but it is not so bad as in my previous comment about this, because the replaced strings look much more like strings that would appear only once in the files..